src/com/github/johnthagen/cppcheck/CppCheckInspectionImpl.java

```diff
@@ -45,72 +55,121 @@ private static ProblemHighlightType severityToHighlightType(@NotNull final Strin
+            */
+            if (id.equals("missingIncludeSystem")) {
+                continue;
+            }
+
+            final String severity = attrs.getNamedItem("severity").getNodeValue();
+            final String errorMessage = attrs.getNamedItem("msg").getNodeValue();
+            final Node inconclusiveNode = attrs.getNamedItem("inconclusive");
+            final boolean inconclusive = inconclusiveNode != null && inconclusiveNode.getNodeValue().equals("true");
+
+            Node location = null;
+
+            // look for the first "location" child name
+            final NodeList children = error.getChildNodes();
+            for(int j = 0; j < children.getLength(); ++j)
+            {
+                final Node child = children.item(j);
+                if (child.getNodeName().equals("location"))
+                {
+                    location = child;
+                    break;
+                }
+            }
+
+            // ignore entries without location e.g. missingIncludeSystem
+            if (location == null) {
                 continue;
             }
 
-            final String fileName = Paths.get(matcher.group(1)).getFileName().toString();
-            int lineNumber = Integer.parseInt(matcher.group(2), 10);
-            final String severity = matcher.group(3);
-            final String errorMessage = matcher.group(4);
+            final NamedNodeMap locattrs = location.getAttributes();
+            final String fileName = new File(locattrs.getNamedItem("file").getNodeValue()).getName();
+            int lineNumber = Integer.parseInt(locattrs.getNamedItem("line").getNodeValue());
+            final int column = Integer.parseInt(locattrs.getNamedItem("column").getNodeValue()); // TODO
 
-            // If a .c or .cpp file #include's header files, Cppcheck will also run on the header files and print
-            // any errors. These errors don't apply to the current .cpp field and should not be drawn. They can
+            // If a file #include's header files, Cppcheck will also run on the header files and print
+            // any errors. These errors don't apply to the current file and should not be drawn. They can
             // be distinguished by checking the file name.
-            // Example:
-            //   Checking Test.cpp ...
-            //   [Test.h:2]: (style) Unused variable: x
-            //   [Test.cpp:3]: (style) Unused variable: y
             if (!fileName.equals(sourceFileName)) {
                 continue;
             }
 
-            // Cppcheck error or parsing error.
+            // Cppcheck error
             if (lineNumber <= 0 || lineNumber > document.getLineCount()) {
```

Comment:
We should be able to remove this in a later release since it should only happen if there's an error without proper location data and I already handled some of those cases.